cpp/src/arrow/io/compressed.cc

```diff
@@ -353,7 +356,9 @@ class CompressedInputStream::Impl {
 
   // Try to feed more data into the decompressed_ buffer.
   Status RefillDecompressed(bool* has_data) {
```

Comment:
Since we are making changes in this function, can you make it return a `Result<bool>`?

Reply:
I don't fully understand, seems that syntax of this function is not changed?

Reply:
Before https://github.com/apache/arrow/pull/39807 . `decompressed_` would set to nullptr if it's all consumed. And even https://github.com/apache/arrow/pull/39807 change the syntax, if would gurantee ok since `DecompressData()` in `compressed_` and `decompressor_` would protect it. So I think this just didn't change the syntax

Reply:
We can just take the opportunity to use the newer style of returning `Result<bool>` instead of taking a `bool*` output argument.

Reply:
Okay, done